test/table.ts

```diff
@@ -1109,350 +1125,154 @@ describe('Bigtable/Table', () => {
-          }) as {} as EventEmitter,
-        ];
-
-        const options = {
-          ranges: [{start: 'a', end: 'c'}],
-          limit: 2,
+        const customCanResume = (error: GoogleError) => error.code === 6;
+        const customGetResumptionRequestFn = (request: RequestType) => {
+          return {fakeProperty: 19};
         };
-
-        callCreateReadStream(options, () => {
-          assert.strictEqual(reqOptsCalls.length, 1);
-          done();
-        });
-      });
-
-      it('should not retry if all the keys are read', done => {
-        emitters = [
-          ((stream: Duplex) => {
-            stream.push([{key: 'a'}]);
-            stream.emit('error', makeRetryableError());
-          }) as {} as EventEmitter,
-        ];
-
-        callCreateReadStream({keys: ['a']}, () => {
-          assert.strictEqual(reqOptsCalls.length, 1);
-          done();
-        });
-      });
-
-      it('shouldn not retry if all the ranges are read', done => {
-        emitters = [
-          ((stream: Duplex) => {
-            stream.push([{key: 'c'}]);
-            stream.emit('error', makeRetryableError());
-          }) as {} as EventEmitter,
-        ];
-
-        const options = {
-          ranges: [{start: 'a', end: 'c', endInclusive: true}],
+        const retry = new RetryOptions(
+          customRetryCodes,
+          customBackOffSettings,
+          customCanResume,
+          customGetResumptionRequestFn
+        );
+        const expectedOptions = {
+          otherArgs: {
+            headers: {
+              'bigtable-attempt': 0,
+            },
+          },
+          retry,
         };
-
-        callCreateReadStream(options, () => {
-          assert.strictEqual(reqOptsCalls.length, 1);
-          assert.deepStrictEqual(reqOptsCalls[0].rows, {
-            rowKeys: [],
-            rowRanges: [{start: 'a', end: 'c', startInclusive: true}],
-          });
-          done();
+        tester.testReadRowsGapicCall(
+          done,
+          {
+            rows: {
+              rowKeys: [],
+              rowRanges: [{}],
+            },
+            tableName,
+          },
+          expectedOptions
+        );
+        const tableWithRetries: Table = bigtable
+          .instance('fake-instance')
+          .table('fake-table');
+        tableWithRetries.createReadStream({
+          gaxOptions: {
+            retry,
+          },
         });
       });
-
-      it('shouldn not retry with keys and ranges that are read', done => {
-        emitters = [
-          ((stream: Duplex) => {
-            stream.push([{key: 'a1'}]);
-            stream.push([{key: 'd'}]);
-            stream.emit('error', makeRetryableError());
-          }) as {} as EventEmitter,
-        ];
-
-        const options = {
-          ranges: [{start: 'a', end: 'b'}],
-          keys: ['c', 'd'],
+      it('should pass the right information to the gapic layer in a complex example', done => {
```

Comment:
What's the "Right information"? request for slightly more detail in naming

Reply:
Updated the description.